gnomad_qc/v4/create_release/create_release_sites_ht_genomes.py

```diff
@@ -690,6 +697,215 @@ def join_release_ht_with_subsets(
+            subset_pop_expr = pop_expr
+
+        # Build strata expressions to get group membership for pop, sex, pop-sex.
+        strata_expr = [
+            {"pop": subset_pop_expr},
+            {"sex": sex_expr},
+            {"pop": subset_pop_expr, "sex": sex_expr},
+        ]
+        # For subsets, add strata expressions to get group membership for subset,
+        # subset-pop, subset-sex, and subset-pop-sex.
+        if subset != "all":
+            subset_expr = hl.or_missing(ht.meta.subsets[subset], subset)
+            strata_expr = [{"subset": subset_expr}] + [
+                {"subset": subset_expr, **x} for x in strata_expr
+            ]
+
+        subset_ht = generate_freq_group_membership_array(
+            ht, strata_expr, remove_zero_sample_groups=True
+        )
+        subset_globals = subset_ht.index_globals()
+        if subset != "all":
+            # The group_membership, freq_meta, and freq_meta_sample_count arrays for
+            # subsets is ordered [adj, raw, subset adj, ...] and we want it to be
+            # [subset adj, subset raw, ...] where subset adj and subset raw have the
+            # same sample grouping.
+            subset_adj_idx = 2
+            subset_ht = subset_ht.annotate(
+                group_membership=hl.array(
+                    [subset_ht.group_membership[subset_adj_idx]]
+                ).extend(subset_ht.group_membership[subset_adj_idx:])
+            )
+            freq_meta_sample_count = hl.array(
+                [subset_globals.freq_meta_sample_count[subset_adj_idx]]
+            ).extend(subset_globals.freq_meta_sample_count[subset_adj_idx:])
+
+            # Modify the freq_meta annotation to match the new group_membership and
+            # freq_meta_sample_count arrays.
+            subset_ht = subset_ht.annotate_globals(
+                freq_meta=hl.array(
+                    [
+                        {"subset": subset, "group": "adj"},
+                        {"subset": subset, "group": "raw"},
+                    ]
+                ).extend(subset_globals.freq_meta[subset_adj_idx + 1 :]),
+                freq_meta_sample_count=freq_meta_sample_count,
+            )
+
+        # Remove 'Han' and 'Papuan' pops from group_membership, freq_meta, and
+        # freq_meta_sample_count.
+        subset_ht = filter_freq_arrays(
+            subset_ht,
+            {"pop": ["han", "papuan"]},
+            keep=False,
+            combine_operator="or",
+            annotations=["group_membership"],
+        )
+
+        # Keep track of which groups should aggregate raw genotypes.
+        subset_ht = subset_ht.annotate_globals(
+            raw_group=subset_ht.freq_meta.map(lambda x: x.get("group", "NA") == "raw"),
+        )
+        hts.append(subset_ht)
+
+    return concatenate_annotations(
+        hts,
+        row_field_names=["group_membership"],
+        global_field_names=list(FREQ_GLOBALS) + ["raw_group"],
+    )
+
+
+def compute_an_by_group_membership(
```

Comment:
one general question: do we need to apply hom_alt_adjustment to these new variants?

Reply:
The `hom_alt_adjustment` doesn't impact AN and that is all we are computing

Reply:
ah sorry, I forgot this.